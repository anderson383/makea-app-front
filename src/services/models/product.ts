import { ICategory } from "./category";

export interface IProduct {
  id: string;
  name: string;
  image: string;
  price: number;
  description: string;
  isAddCart?: boolean;
  categoria?: string | ICategory;
}