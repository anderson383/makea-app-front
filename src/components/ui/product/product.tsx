import { View, Text, StyleSheet, FlatList, ScrollView, Image, Button, TouchableNativeFeedback, TouchableOpacity, Touchable } from "react-native"
import { THEME } from "./../../../theme";
import { SvgXml } from 'react-native-svg';
import { IProduct } from "../../../services/models/product";
import { numberFormat } from "../../../helpers/numberCurrency";
import {HeartIcon, HeartIconActive} from "../../../../assets/icons/heart";

export interface ProductProps {
  product: IProduct
  onClickHeart?: () => void;
  onClickCard?: () => void;
}

const Product:React.FC<ProductProps> = ({product, onClickHeart, onClickCard}) => {
  return (
      <View style={styles.product} key={product.id}>
        <TouchableOpacity onPress={onClickCard && onClickCard} testID="productCard">
          <View style={{flex: 1}}>
            <Image source={{ uri: product.image, height: 150 }} style={{width: '100%'}} />
          </View>
          <View style={styles.info} >
            <View style={styles.priceContent}>
              <Text style={styles.price}>{numberFormat(product.price)}</Text>
              <TouchableOpacity  onPress={onClickHeart && onClickHeart} testID="addCart">
                { product.isAddCart ?  <SvgXml  xml={HeartIconActive()}  /> : <SvgXml  xml={HeartIcon()}  />} 
              </TouchableOpacity>
            </View>
            <View>
            <Text>{product.name}</Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
   
  )
}

const styles = StyleSheet.create({
  product: {
    marginRight: 3,
    borderRadius: 10,
    flexBasis: '47.8%',
    overflow: 'hidden',
    borderColor: '#f1f1f1',
    borderWidth: 1
  },
  info: {
    padding: 8,
    position: 'relative',
    gap: THEME.paddings.sm
  },
  priceContent: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  price: {
    color: THEME.colors.secondaryColor,
    fontWeight: 'bold'
  }
})

export default Product;