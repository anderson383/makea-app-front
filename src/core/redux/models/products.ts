import { IProduct } from '../../../services/models/product';

export interface IProductState {
  products: IProduct[]
}