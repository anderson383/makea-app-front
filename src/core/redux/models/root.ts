import { IProductState } from "./products";

export interface IrootState {
  product: IProductState
}