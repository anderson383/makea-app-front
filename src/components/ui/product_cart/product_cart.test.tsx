
import ProductCart from '.'
import {render, fireEvent, waitFor} from '@testing-library/react-native'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { IProduct } from '@/src/services/models/product';


const productMock = {
  id: 'id-test',
  image: 'imgaentest',
  description: 'product description',
  price: 2000,
  categoria: 'category',
  isAddCart: true,
  name: 'Chicharron! xd'
}

const mockStore = configureStore([]);

describe('Test shared ProductCart Component', () => {
  
  const setup = async (store:any, product:IProduct) => {
    return await waitFor(() => render(
      <Provider store={store}>
        <ProductCart product={product} />
      </Provider>
    )) 
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test(`Should render component`, async () => {
    const store = mockStore({ product: { products: [] } })

    const {toJSON} = await setup(store, productMock)

    expect(toJSON).toMatchSnapshot()
  })

  test(`Should delete item`, async () => {
    const store = mockStore({ product: { products: [] } })

    const {getByTestId} = await setup(store, productMock)

    const butt = getByTestId('deleteItem')
    
    fireEvent.press(butt)

    expect(store.getActions()[0].payload.isAddCart).toBe(!productMock.isAddCart)
  })   
})
