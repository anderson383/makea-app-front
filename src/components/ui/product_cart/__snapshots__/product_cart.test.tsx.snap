// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test shared ProductCart Component Should render component 1`] = `[Function]`;
